### ISSUE TITLE
Mojito is not re-using existing translations for Android plurals

### ISSUE THREAD

**Issue by ehoogerbeets** [box/mojito#291]
We found a situation where the old mojito (version 0.81) is generating the correct translations for plurals in an android app, but the latest mojito is not. (Version 0.84 and also the head of the master branch)

What is happening is that the latest mojito is generating all of the plural classes for various languages. This is a good thing because some of those classes were missing before. It is producing those strings with the English text because the translations are not there, which is also good. The part that is wrong is that the old mojito generated the strings for the "one" and "other" classes with the correct translations in them, but the new mojito only generates the string for the "one" class with the correct translation. The "other" string is in English, even though the translation is available.

I have distilled it down to a minimal set of files which I tarred up and put on my Box account publically: https://app.box.com/s/u9o7d0jg8hspe2p6h6yl4i1v02ose2na

The "bug.sh" script shows the commands I used to repro the problem. The source/res/values/strings.xml is the source file with just a single plural array in it. The tm/\*.xliff are the translation memory files with the translations for Arabic, Polish, Russian, and Chinese. The localized/res/\*/strings.xml files are the output files from running that script with mojito 0.84. You can see that the "other" classes are missing the translations that exist in the TMs for each of those locales.

**Comment by aurambaj** [comment]
@ehoogerbeets the xliff files don't look good. 'ar-SA' is missing some forms and 'zh' has too many though this shouldn't hurt. 

If I remember well I didn't really look into the TM import/export when I implemented the plural support so it might have never worked. 

Are the drops working fine with plural (is putting the right amount of forms for each language)? Can you confirm that plural support is ok in the workbench and generated files when not using the TM import/export?

**Comment by aurambaj** [comment]
I don't remember exactly the timeline of the releases and I thought you were on an higher version but there might be something with the `PluralFormUpdater` not ever being ran. 

I made it optional and disabled by default it at some point and it is possible it was never run based on your version. 

I'm reintroducing that as something to be run only once with the Quartz Scheduler that I'll push soon. but for now you can enable it with: `l10n.PluralFormUpdater=true` property. 

If ran for the first time it does changes to the db so that plural support works fine. Could you check if that helps?

**Comment by ehoogerbeets** [comment]
Sure, I'll check it and get back to you.

**Comment by ehoogerbeets** [comment]
Actually, we are testing with HSQL so the PluralFormUpdater won't be run:

PluralFormUpdate.java:44
```
        } else if ("HSQL".equals(driver)) {
           logger.debug("Don't update (DB is HSQL)");
```

**Comment by aurambaj** [comment]
ok, plus disregard what i said. the `PluralFormUpdater` should keep running until all repositories are migrated. One time execution is not good, I need to fix the change made to Quartz and I'll enable it by default. To prevent it to run it will have to be explicitly defined in the config. It will make things safer.

I'll have to debug to see what's going with your issue. Can't say much more like that.

**Comment by aurambaj** [comment]
I think I know what's going on, it is a little messy:

```
<!-- Delete -->
    <plurals name="numberOfDeleteFolders">
        <item quantity="one" description="Confirmation to delete a single folder e.g. 'Are you sure you want to delete the folder 'Work' and all its contents?">Are you sure you want to delete the folder &lt;b>\'%2$s\'&lt;/b> and all of its contents?</item>
        <item quantity="other" description="Confirmation to delete more than one folder" >Are you sure you want to delete these %1$d folders?</item>
    </plurals>
```

The Android filter accept 2 ways of giving translator directives:
- XML comment before the text unit 
- "description" attribute in the XML element

"description" attribute doesn't seem to be extracted for plural forms, that probably only works for simple entries. So `Delete` is used a source comment. (This is probably not what you want in your case since the developer gave good comments in "description" but anyway that's another story)

So the problem is that in the TM export the "other" form doesn't have a source comments. only the "one" form has it ("Deleted"). Probably due to the older behavior of the filter.

The current filter make it so that the source comment is the same for all forms. So when pushing the source file, it creates text unit for all forms with "Deleted" source comment. The it finds an exact match for "one" but doesn't for "other". 

If you look at unused translation you'll find the old string for "other" with the old translations actually imported. I would have expected the leveraging to kick in and copy the translations... That must be some edge case again.
